From e82eb3731d282f137b553c4bd5e52305e967ee5d Mon Sep 17 00:00:00 2001
From: Thomas Preud'homme <thomas.preudhomme@celest.fr>
Date: Tue, 13 Apr 2010 21:08:37 +0200
Subject: Error out when declaring a static function without file scope

Error out when declaring a static function in another scope than file scope, as
required by C99 6.2.2:21

Origin: vendor
Bug-Debian: http://bugs.debian.org/170105
Forwarded: http://lists.nongnu.org/archive/html/tinycc-devel/2011-02/msg00047.html
Last-Update: 2011-02-09
Applied-Upstream: commit: c23400278ad7a1a44cb9b469c4635ea29cb12b25
---
 tccgen.c |    3 +++
 1 files changed, 3 insertions(+), 0 deletions(-)

diff --git a/tccgen.c b/tccgen.c
index 5fa9f43..fee07ae 100644
--- a/tccgen.c
+++ b/tccgen.c
@@ -4998,6 +4998,9 @@ static void decl(int l)
             }
 #endif
             if ((type.t & VT_BTYPE) == VT_FUNC) {
+                if ((type.t & VT_STATIC) && (l == VT_LOCAL)) {
+                    error("function without file scope cannot be static");
+                }
                 /* if old style function prototype, we accept a
                    declaration list */
                 sym = type.ref;
-- 
1.7.5.4

